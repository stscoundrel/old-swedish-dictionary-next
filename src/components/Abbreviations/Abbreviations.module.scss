.column {
  margin-bottom: 30px;
}

.wrap {
  display: flex;
}

.abbreviations {
  flex-wrap: wrap;
}

.abbreviation {
  flex: 1;
  position: relative;
  padding-right: 20px;

  &::after {
    content: 'â†’';
    position: absolute;
    right: 50px;
    top: 0;
  }
}

.explanation {
  flex: 2;
}

.link {
  flex-basis: 100%;
}


@media only screen and (min-width: 768px) {
  .abbreviations {
    display: flex;
    margin-top: 50px;
  }

  .column {
    flex-basis: 70%;
    width: 70%;
  }

  .explanation {
    flex: 3;
  }
}