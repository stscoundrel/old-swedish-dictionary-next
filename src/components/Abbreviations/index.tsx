import { Abbreviation } from 'lib/services/abbreviations'
import Link from 'next/link'
import styles from './Abbreviations.module.scss'

interface AbbreviationProps{
  abbreviations: Abbreviation[]
}

export default function Abbreviations({ abbreviations }: AbbreviationProps) {
  const shouldShowAllAbbreviationsLink = (): boolean => {
    if (abbreviations.length > 200) {
      // Sources index, no need to link to the same page.
      return false
    }

    if (abbreviations.length > 0) {
      return true
    }

    return false
  }

  return (
    <div className={styles.abbreviations}>
      {abbreviations.length > 0
        && <div className={styles.column}>
          <h4>Works and authors cited:</h4>
          {abbreviations.map(({ abbreviation, explanation }) => (
            <dl className={styles.wrap} key={abbreviation}>
              <dt className={styles.abbreviation}>
                <strong>{abbreviation}</strong>
              </dt>
              <dd className={styles.explanation}>{explanation}</dd>
            </dl>
          ))}
        </div>
      }

      {shouldShowAllAbbreviationsLink()
      && <Link href="/source-list" className={styles.link} prefetch={false}>
       âžž See all works cited in the dictionary
      </Link>
      }
    </div>
  )
}
