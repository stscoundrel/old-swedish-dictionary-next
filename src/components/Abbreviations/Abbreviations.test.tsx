import React from 'react'
import renderer from 'react-test-renderer'
import Abbreviations from './index'

describe('Abbreviations component', () => {
  const abbreviations = [
    {
      abbreviation: 'GU',
      explanation: 'Glossarii Latino-Svethici specimen vetustum. E cod. mscr. Bibliothecæ Reg. Acad. Upsal. Diss. Ups. præs. J. H. Schröder. 1845.',
    },
    {
      abbreviation: 'GU C 20',
      explanation: 'Latinskt-Svenskt glossarium efter Cod. Ups. C 20. Utg. av E. Neuman. S 1--583. 1918--20, (hand 2) s. 1--169. 1938--42. SFSS.',
    },
  ]

  test('Matches the snapshot', () => {
    const tree = renderer.create(
      <Abbreviations abbreviations={abbreviations} />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
