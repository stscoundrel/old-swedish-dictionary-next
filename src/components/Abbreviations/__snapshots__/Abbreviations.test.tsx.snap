// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Abbreviations component Matches the snapshot 1`] = `
<div
  className="abbreviations"
>
  <div
    className="column"
  >
    <h4>
      Works and authors cited:
    </h4>
    <dl
      className="wrap"
    >
      <dt
        className="abbreviation"
      >
        <strong>
          GU
        </strong>
      </dt>
      <dd
        className="explanation"
      >
        Glossarii Latino-Svethici specimen vetustum. E cod. mscr. Bibliothecæ Reg. Acad. Upsal. Diss. Ups. præs. J. H. Schröder. 1845.
      </dd>
    </dl>
    <dl
      className="wrap"
    >
      <dt
        className="abbreviation"
      >
        <strong>
          GU C 20
        </strong>
      </dt>
      <dd
        className="explanation"
      >
        Latinskt-Svenskt glossarium efter Cod. Ups. C 20. Utg. av E. Neuman. S 1--583. 1918--20, (hand 2) s. 1--169. 1938--42. SFSS.
      </dd>
    </dl>
  </div>
  <a
    className="link"
    href="/source-list"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    ➞ See all works cited in the dictionary
  </a>
</div>
`;
